writings/python-api-design.md

```diff
@@ -458,38 +456,72 @@ def order(chips=None, fish=None):
 
 Using `ValueError` is fine here -- there's no *value* in defining a custom exception type.
 
-However, if your error needs to provide additional information, a custom `Exception` subclass is useful. For example:
+If you're creating something that maps very well to standard operations, like a filesystem library, you might want to reuse [`OSError` subclasses](https://docs.python.org/3/library/exceptions.html#os-exceptions) like [`FileNotFoundError`](https://docs.python.org/3/library/exceptions.html#FileNotFoundError) or [`PermissionError`](https://docs.python.org/3/library/exceptions.html#PermissionError).
+
+However, when you're building a new library, it's usually best to create a custom class for all of your library's exceptions, and raise meaningful subclasses of this. Your base class should inherit from `Exception`. For example, in our fish ’n’ chips library, you might do:
 
 ```python
-class OrderError(Exception):
-    def __init__(self, shop_url, chef_name, message):
-        self.shop_url = shop_url
+class Error(Exception):
```

Comment:
That is certainly fine. I think 
```suggestion
class <name-of-library>Error(Exception):
```
is even clearer - but not sure how to present that.

Maybe 

```
class MyLibraryError(Exception):
```
?

Reply:
Could also just clarify that the intention in this example is that these exceptions are imported along with the main lib module as `import fishnchips` and intended to be used name-spaced such as `except fishnchips.Error:`.

Reply:
Per my naming section, I'm a big believer in using names with the package namespace, so `fishnchips.Error` as Mitch pointed out, rather than the repetition and awkwardness of `fishnchips.FishNChipsError`. However, I've added a couple of comments to clarify that.